pkg/usersettings/handlers.go

```diff
@@ -0,0 +1,340 @@
+				},
+				Verbs: []string{
+					"get",
+					"list",
+					"patch",
+					"update",
+					"watch",
+				},
+				ResourceNames: []string{
+					configMapName,
+				},
+			},
+		},
+	}
+
+	roleBinding := RoleBinding{
+		ApiVersion: "rbac.authorization.k8s.io/v1",
+		Kind:       "RoleBinding",
+		Metadata: Metadata{
+			Namespace: namespaceName,
+			Name:      roleBindingName,
+		},
+		Subjects: []RoleBindingSubject{
+			RoleBindingSubject{
+				ApiGroup: "rbac.authorization.k8s.io",
+				Kind:     "User",
+				Name:     user.Username,
+			},
+		},
+		RoleRef: RoleRef{
+			ApiGroup: "rbac.authorization.k8s.io",
+			Kind:     "Role",
+			Name:     roleName,
+		},
+	}
+
+	configMap := ConfigMap{
+		ApiVersion: "v1",
+		Kind:       "ConfigMap",
+		Metadata: Metadata{
+			Namespace: namespaceName,
+			Name:      configMapName,
+		},
+	}
+
+	roleResponse, err := h.createRole(&role)
+	if err != nil {
+		fmt.Printf("createRole error: %v", err)
+	} else {
+		fmt.Printf("createRole response: %s", h.toString(roleResponse))
+	}
+
+	roleBindingResponse, err := h.createRoleBinding(&roleBinding)
+	if err != nil {
+		fmt.Printf("createRoleBinding error: %v", err)
+	} else {
+		fmt.Printf("createRoleBinding response: %s", h.toString(roleBindingResponse))
+	}
+
+	configMapResponse, err := h.createConfigMap(&configMap)
+	if err != nil {
+		fmt.Printf("createRoleBinding error: %v", err)
+	} else {
+		fmt.Printf("createRoleBinding response: %s", h.toString(roleBindingResponse))
+	}
+
+	if err != nil {
+		return nil, err
+	} else if configMapResponse.StatusCode == 409 {
+		return h.getConfigMap(namespaceName, configMapName)
+	} else {
+		return configMapResponse, nil
+	}
+}
+
+func (h *UserSettingsHandler) deleteUserSettings(user *auth.User) (*http.Response, error) {
+	resourceNameSuffix, err := h.getResourceNameSuffix(user)
+	if err != nil {
+		return nil, err
+	}
+
+	namespaceName := h.getNamespaceName()
+	configMapName := h.getConfigMapName(resourceNameSuffix)
+	roleName := h.getRoleName(resourceNameSuffix)
+	roleBindingName := h.getRoleBindingName(resourceNameSuffix)
+
+	h.deleteRoleBinding(namespaceName, roleBindingName)
+	h.deleteRole(namespaceName, roleName)
+	return h.deleteConfigMap(namespaceName, configMapName)
+}
+
+func (h *UserSettingsHandler) getCurrentUser(user *auth.User) (*User, error) {
+	path := fmt.Sprintf("/apis/user.openshift.io/v1/users/~")
+	url := proxy.SingleJoiningSlash(h.Endpoint, path)
+
+	request, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	request.Header.Set("Authorization", fmt.Sprintf("Bearer %s", user.Token))
+	request.Header.Set("Accept", "application/json")
+
+	response, err := h.Client.Do(request)
+	if err != nil {
+		return nil, err
+	}
+
+	fmt.Printf("getCurrentUser %d %s\n", response.StatusCode, response.Status)
+
+	if response.StatusCode != 200 {
+		return nil, errors.New(fmt.Sprintf("Unexpected status code (%d instead of 200) when getting current user", response.StatusCode))
```

Comment:
Log this error.

Reply:
Outdated since I switched to client-go, also added some klogs